espflash/src/chip/esp32.rs

```diff
@@ -49,7 +55,31 @@ impl ChipType for Esp32 {
     ) -> Box<dyn Iterator<Item = Result<RomSegment<'a>, Error>> + 'a> {
         let bootloader = include_bytes!("../../bootloader/esp32-bootloader.bin");
 
-        let partition_table = PartitionTable::basic(0x10000, 0x3f0000).to_bytes();
+        let partition_table = PartitionTable::new(
```

Comment:
Might also want to do this for the c3?

Reply:
Seems it's the exact same layout
https://docs.espressif.com/projects/esp-idf/en/latest/esp32c3/api-guides/partition-tables.html

Reply:
Might make sense to just edit it in `partition_table.rs`, because it's the same for the s2 as well

Reply:
Ah, for some reason I thought the c3 was different -- at least it might be sometimes?  I remembered seeing this: https://github.com/ivmarkov/rust-esp32-std-hello/#flashing-for-esp32-s2--esp32-c3

But if the s2 is the same, at least, makes sense to put it somewhere common.  And either way, I guess this 3-part table is perhaps more useful for all boards than the basic one there now.

Reply:
Anyhow, this is done